web-console/4.0-designs/storage/Clone/clone.md

```diff
@@ -0,0 +1,55 @@
+# Clone - PVC/PV
+
+## Main use cases
+Create clone (from where PVC/snapshot is)
+All other List, View, Delete, Modify actions are that of a PV/PVC
+
+#### Notes:
+- Alpha support in Kubernetes v1.15
+- critical for stateful workloads, especially databases
+- similar behavior to creating a snapshot
+- From user’s perspective, a clone is another PVC/PV; however contents of a clone is that its PV is populated from contents of a PV at creation time.
+- After creation, the clone behaves just like any other PV.
+- clone using create volume from existing snapshot or existing PV
+- Cloning only support with CSI drivers (and not for in-tree or Flex drivers)
+- To enable the Kubernetes Volume Cloning feature 
+- enabled via the PersistentVolumeClaim.DataSource.Kind field
+ 
+## User Stories
+As a Developer, I want to be able to create a clone of a PVC.
+- Name - suggest a default, e.g. clone-timestamp
+- Select source PVC
+- Requested size (this capacity must be specified and must be >= the capacity of the source volume)
+- Scheduling support
+- Note: The Create Clone action should only be exposed for PVCs backed by supported CSI drivers that support cloning.
+ 
+#### Notes:
+- Cloning action is only presented for drivers that support cloning.
+- This may only apply to ceph-rbd (block-based PVs and snapshots).
+
+
+## PVC/PV Clone Workflow
+### List Page
+![PVC List Pgae](img/PVC-List-Clone-01.png) 
+
+Clicking on the clone option in the kebab menu will open the clone modal
+![PVC List Pgae](img/PVC-List-Clone-02.png) 
+
+#### Clone Modal
+The clone modal includes a field for name and the pvc details. 
+![PVC List Pgae](img/PVC-List-Clone-03.png) 
```

Comment:
Should this be in a full-page form? I think the design convention is that if it's a form of larger than 2-3 fields it would be consistent to do full page. Maybe worth checking in with @rileyhuston on that though.